qiita_db/metadata_template/prep_template.py

```diff
@@ -610,20 +603,11 @@ def qiime_map_fp(self):
         str
             The filepath of the QIIME mapping file
         """
-        with qdb.sql_connection.TRN:
-            sql = """SELECT filepath_id, filepath
-                     FROM qiita.filepath
-                        JOIN qiita.{0} USING (filepath_id)
-                        JOIN qiita.filepath_type USING (filepath_type_id)
-                     WHERE {1} = %s AND filepath_type = 'qiime_map'
-                     ORDER BY filepath_id DESC""".format(self._filepath_table,
-                                                         self._id_column)
-            qdb.sql_connection.TRN.add(sql, [self._id])
-            # We know that the good filepath is the one in the first row
-            # because we sorted them in the SQL query
-            fn = qdb.sql_connection.TRN.execute_fetchindex()[0][1]
-            base_dir = qdb.util.get_mountpoint('templates')[0][1]
-            return join(base_dir, fn)
+        for _, fp, fp_type in qdb.util.retrieve_filepaths(
+                self._filepath_table, self._id_column, self.id,
+                sort='descendent'):
+            if fp_type == 'qiime_map':
+                return fp
```

Comment:
That case should never happen (i.e. all prep template objects should have the qiime_map).
However, in case that it doesn't exist, I think returning None is the correct approach.